library/src/main/java/com/yayandroid/locationmanager/providers/locationprovider/GooglePlayServicesLocationSource.java

```diff
@@ -89,17 +89,23 @@ void startSettingsApiResolutionForResult(Status status, Activity activity) throw
 
     @SuppressWarnings("ResourceType")
     void requestLocationUpdate() {
-        LocationServices.FusedLocationApi.requestLocationUpdates(googleApiClient, locationRequest, this);
+        if (isGoogleApiClientConnected()) {
```

Comment:
I see you log on issue, in that case, do the check [there](https://github.com/yayaa/LocationManager/blob/master/library/src/main/java/com/yayandroid/locationmanager/providers/locationprovider/GooglePlayServicesLocationProvider.java#L236) not here. And only call `connect` don't ask for location again, provider will handle when it is connected.